### ISSUE TITLE
Sort local playlists in groups / folders

### ISSUE THREAD

**Issue by justanidea** [TeamNewPipe/NewPipe#1313]
It could be cool to sort our local playlist, because when you have a lot of it its kinda messy.
If someone would like to do that, I would advise him/her not to do folders, but things more like the animation used for "import export" in the sub page for each new category created by the user.

**Comment by Stypox** [comment]
Related: https://github.com/TeamNewPipe/NewPipe/issues/1982#issuecomment-528029200

**Comment by MD77MD** [comment]
@Stypox https://github.com/TeamNewPipe/NewPipe/issues/1313#issuecomment-528029306... although I'm a fan of tags...I agree with you, folders make more sense here and alot easier to use too.

This would help organize all my playlist.. a special folder for all my music 🎶 playlists... I can even have it divided by artist, now that's cool 😎

**Comment by justanidea** [comment]
Damn, the me of 2 years ago had such a good idea :o And @MD77MD is on the same direction as me, i was thinking in sorting my musics too when writing this post (i usually get 1000 more musics each year...)

**Comment by MD77MD** [comment]
this is the kind of small features which reviels how stupid and manipulative YouTube is -aka Google- and once some tires them, can never go back to the YouTube app

**Comment by cntrl-s** [comment]
is someone working on this? 

i want to try to implement this

**Comment by Stypox** [comment]
No, nobody is working on this at the moment, so feel free to pick it up! Thank you :-D
Before taking ui / usability design decisions I suggest you to get some feedback here, so that you won't have to make more changes later ;-)

**Comment by cntrl-s** [comment]
> Before taking ui / usability design decisions I suggest you to get some feedback here, so that you won't have to make more changes later ;-)

i will, thank you :)

**Comment by MD77MD** [comment]
any update?

i don't know coding but I can help with testing and feedback.

**Comment by cntrl-s** [comment]
@MD77MD i am new to android development so this will take a while 

i will let you know once i have something for you to test :P

**Comment by cntrl-s** [comment]
@MD77MD @Stypox how do you think we should implement playlist group view 

i am thinking horizontally scrollable (like feed group) under playlist tab with thumbnail of first playlist under the group as thumbnail for the group

**Comment by XiangRongLin** [comment]
@cntrl-s  I would do it like the channel groups/subscription groups.
So the current playlist view gets expanded by a header to manage the groups. When clicking on one of those groups it opens the old playlist overview with all the playlists in that group. From there you can open a single playlist to see the single videos.

It would allow to reuse/copy stuff, depending if you want to extract out common components or just copy over what you need. Restriction would be that you can only have 1 layer and thus can't have nested groups, like a traditional file system.
But i dare say that multilayer approach is a lot harder and not necessary.

**Comment by cntrl-s** [comment]
>  But i dare say that multilayer approach is a lot harder and not necessary.

@XiangRongLin i agree with this, i was thinking we could have simple groups like subscription groups

**Comment by JeffRockatansky** [comment]
JSON / YAML import / export of playlists feature would also be nice.

And maybe an archiving feature per playlist?

Guess I'll describe these ideas in separate issues, but they should be considered in possible playlists tab redesign.

**Comment by Stypox** [comment]
>  i agree with this, i was thinking we could have simple groups like subscription groups

I also agree, but I'd use the group's thumbnail as a background image for the group item. Also, I'd make the playlist group items have: a height that is something in between 1.5x and 2x the height of subscription group items; a width/height ratio of 16:9.

> Guess I'll describe these ideas in separate issues,

@JeffRockatansky yeah, those are different and unrelated proposals

**Comment by cntrl-s** [comment]
> @cntrl-s I also agree, but I'd use the group's thumbnail as a background image for the group item. 

that's what i said in a reply above we could use the thumbnail of the first playlist under the group as thumbnail for the group or we could combine different playlists thumbnail to make a new one (vlc android does this for folder with multiple videos)

**Comment by Stypox** [comment]
@cntrl-s if this does not require storing too much cached data on the user's phone then ok

**Comment by cntrl-s** [comment]
@Stypox ok i will start working on this, thank you

**Comment by Stypox** [comment]
@cntrl-s thanks to you! :-D

**Comment by cntrl-s** [comment]
@JeffRockatansky @Stypox @MD77MD @XiangRongLin how do you think the playlist selection view should be when selecting playlists for a group

should every playlist have a radio button or have a '✔' on top of selected playlists or something else

**Comment by XiangRongLin** [comment]
I personally don't care too much about it. Try to make it simple and close to the existing channel groups.

So a list where if one entry is selected the thumbnail is covered my a ✔，would be the simplest that i can think of right now

**Comment by JeffRockatansky** [comment]
@cntrl-s a tick over a playlist is good. Radio will take too much additional space.

Maybe you could also try implementing paths while naming a playlist? E.g. when you type `Parent/Child` or `Parent.Child` (or with any other custom delimiter) then playlist `Child` is moved into playlist `Parent`, so the tree could look like this:
```
Parent
├ Child
│ └ [videos]
└ [videos]
Aunt
├ Cousin
│ └ [videos]
└ [videos]
```
Parsing existing names would also be very much appreciated, especially because my playlists all look like `_sw.os.linux.arch`, `_sci.phys.quant` etc.

**Comment by XiangRongLin** [comment]
@JeffRockatansky Im going to give this a hard NO, because this is complex enough as is.

Please open a seperate issue if you want the feature.

**Comment by JeffRockatansky** [comment]
@XiangRongLin ok, I get it.

What about the structure of folders itself? Should A folder contain both videos and sublists or only sublists?

**Comment by XiangRongLin** [comment]
https://github.com/TeamNewPipe/NewPipe/issues/1313#issuecomment-767414178

**Comment by JeffRockatansky** [comment]
@XiangRongLin could you explain in detail why multilayer is harder? And why do you think it's not necessary?

**Comment by XiangRongLin** [comment]
> 
> 
> @XiangRongLin could you explain in detail why multilayer is harder? And why do you think it's not necessary?

Because its __multiple__ layers, meaning more stuff to do. More changes need to be made to the data model, the logic and the representation. More needs to be tested and more code needs to be reviewed. Overall its more work, which can be split up into multiple feature request and multiple PR

**Comment by cntrl-s** [comment]
@JeffRockatansky i will implement it with dim and a tick

**Comment by cntrl-s** [comment]
> Because its multiple layers, meaning more stuff to do. More changes need to be made to the data model, the logic and the representation. More needs to be tested and more code needs to be reviewed. Overall its more work, which can be split up into multiple feature request and multiple PR

also we have to keep track of both local and remote playlists within the folder structure which would be very complex to design database for and this would be even harder as sqlite doesn't support constraints on the database table, we probably would have to keep track of local and remote playlists separately

**Comment by Sam4000der3** [comment]
I'm still hope for this feature. I have to much playlists. I hope to filter my playlists by genre (music playlists) and language.

**Comment by MD77MD** [comment]
any luck with this?

**Comment by Sam4000der3** [comment]
Noop. Unfortunately not

**Comment by Sam4000der3** [comment]
> any luck with this?

Noop. Unfortunately not

**Comment by RohitSingh107** [comment]
Any progress on this?

**Comment by mhdasm** [comment]
This feature is a must, try to implement it

Multi-layer sorting will be very useful for playlists
